### PR TITLE
Add experimental Debezium SMT

### PR DESCRIPTION
**Pull request by bryanck**
This PR adds an experimental Debezium SMT that can be used to ingest data from a Debezium source.

### CODE REVIEW

File: kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/CopyValue.java [hunk 1]
```diff
@@ -81,24 +81,24 @@ public R apply(R record) {
   }
 
   private R applySchemaless(R record) {
-    final Map<String, Object> value = Requirements.requireMap(record.value(), "copy value");
+    Map<String, Object> value = Requirements.requireMap(record.value(), "copy value");
```
Comment (danielcweeks) [inline kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/CopyValue.java:84]: Is removing `final` here necessary?  Seems like an unrelated change.

File: kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/DebeziumTransform.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  }
+
+  @SuppressWarnings("unchecked")
+  private R applySchemaless(R record) {
+    Map<String, Object> value = Requirements.requireMap(record.value(), "Debezium transform");
+
+    String op = mapOperation(value.get("op").toString());
+
+    Object payload;
+    if (op.equals(CdcConstants.OP_DELETE)) {
```
Comment (danielcweeks) [inline kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/DebeziumTransform.java:96]: I was just looking at some update examples that appear to have both `before` and `after`, but this doesn't look like it would support that.  Just wonder if there's a case for both.

**Review by danielcweeks — APPROVED** [review]

File: kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/DebeziumTransform.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+      payload = value.get("before");
+    } else {
+      payload = value.get("after");
+    }
+
+    if (!(payload instanceof Map)) {
+      LOG.debug("Unable to transform Debezium record, payload is not a map, skipping");
```
Comment (amogh-jahagirdar) [inline kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/DebeziumTransform.java:103]: Is this something that should be warn level? or are there any common cases where it's not a map, and it would just be too noisy?

File: kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/CopyValue.java (hunk 1 above)
Comment (bryanck) [reply kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/CopyValue.java:84]: Yeah, I was cleaning some things up I noticed, this is unrelated (should have mentioned that)

File: kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/CdcConstants.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.tabular.iceberg.connect.transforms;
+
+public interface CdcConstants {
```
[the remark this answers is not in the document]
Comment (bryanck) [reply kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/CdcConstants.java:21]: It makes it less verbose and ensures the members are all public static final. Also you could optionally implement without having to extend (though that probably is less useful).

File: kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/DebeziumTransform.java (hunk 3 above)
Comment (bryanck) [reply kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/DebeziumTransform.java:103]: I had it at warn originally, but I wanted to avoid warn for per-message logs, which could explode logging

File: kafka-connect-transforms/README.md [hunk 5]
```diff
@@ -37,3 +37,17 @@ The SMT currently has no configuration. It can be used with the sink's CDC featu
...
+
+The `DebeziumTransform` SMT transforms a Debezium formatted message for use by the sink's CDC feature.
+It will promote the `before` or `after` element fields to top level and add the following metadata fields:
+`_cdc_op`, `_cdc_ts`, and `_cdc_table`.
+
+## Configuration
+
+The SMT currently has no configuration. It can be used with the sink's CDC feature, e.g.
+```
+"iceberg.tables.cdcField": "_cdc_op",
```
[the remark this answers is not in the document]
Comment (bryanck) [reply kafka-connect-transforms/README.md:52]: No it is not configurable in either, Debezium has a message spec that defines their format, and here we don't have a spec but just a convention.

[the remark this answers is not in the document]
Comment (ajreid21) [reply kafka-connect-transforms/README.md:52]: sure, as far as "Debezium has a message spec that defines their format" -- we are already transforming the dbz payload into a table structure (where in the payload there is an "op" field and we are promoting that into "_cdc_op" in the table (this would just be configuring what the table column to promote that "op" field into would be)

[the remark this answers is not in the document]
Comment (bryanck) [reply kafka-connect-transforms/README.md:52]: This config is mostly for other cases when not using the transforms we're providing

File: kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/DebeziumTransform.java (hunk 2 above)
Comment (bryanck) [reply kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/DebeziumTransform.java:96]: For inserts or updates we want `after`. For deletes, after will be null, so we pick `before`. Did I understand the question right?

File: kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/CdcConstants.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (bryanck) [reply kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/CdcConstants.java:21]: It looks like Iceberg doesn't use this convention so I may need to change that if we end up submitting this to Iceberg.

**Review by amogh-jahagirdar — APPROVED** [review]
